### PR TITLE
Fix text inside parentheses disappears from action tooltip

### LINKED ISSUE #59914: Registering an action in the ShortcutManager discards any text inside parenthesis in the tooltip
### What is the bug or the crash?

See https://github.com/qgis/QGIS/pull/59676#issuecomment-2544261523

In order to dynamically change the tooltip when a new shortcut is assigned, any text inside parenthesis is removed from the tooltip, and ` (<MY_NEW_SHORTCUT>)` is appended at the end.

### Steps to reproduce the issue

- Create an action with some text inside parenthesis in its tooltip. 
- Register it in the ShortcutManager
- Assign a new shortcut
- Text inside parentesis disappears

### Versions

3.40

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [X] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by YoannQDQ**
- Fix #59914

Instead of removing text inside parenthesis anywhere in the tooltip when trying to replace the shortcut look only at last occurrence of a text inside parentheses, and check if it is actually a shortcut, and not some informative text.

### CODE REVIEW

**Review by troopa81 — COMMENTED** [review]
It would still fail if someone puts something between parenthesis at the end of the tooltip. Shall we not register the tooltip in mActions to generate the final tooltip text (the one with shortcut sequence) in updateActionToolTip to avoid this fragile code?

**Comment by YoannQDQ** [comment]
Yes, ideally that would be more robust. Every tooltip that currently includes a shortcut would have to be retranslated though.

But unless I'm mistaken, this PR handles the "text inside parenthesis at the end" situation, by checking the text is a valid QKeySequence

**Comment by troopa81** [comment]
> But unless I'm mistaken, this PR handles the "text inside parenthesis at the end" situation, by checking the text is a valid QKeySequence

Maybe. Do you confirm that an invalid QKeySequence would produce an empty QKeySequence ? I would have said that QKeySequence("nonsense") would have returned a isEmpty() false, but a count() == 0, no ?

Once we figured out this, I'm ok with the proposal

**Comment by YoannQDQ** [comment]
Actually, an invalid key sequence would not necessarily produce an empty string. The input string is split by comma, then every group is separetely evaluated, and joined in a final string around `", "`.
The sequence `"gibberish,Ctrl+A,B,garbage"` would hence produce `", Ctrl+A, B, "`. `count()` just returns the number of groups (4 in this example, 1 for `QKeySequence("nonsense")`)

I updated the PR accordingly

File: src/gui/qgsshortcutsmanager.cpp [hunk 1]
```diff
@@ -369,9 +369,26 @@ void QgsShortcutsManager::shortcutDestroyed( QShortcut *shortcut )
...
+    bool validSequence = true;
+    for ( const QString &part : QKeySequence( match.captured( 1 ) ).toString().split( "," ) )
+    {
+      if ( part.trimmed().isEmpty() )
+      {
+        validSequence = false;
+        break;
+      }
+    }
+    if ( validSequence )
```
Comment (troopa81) [inline src/gui/qgsshortcutsmanager.cpp:387]:
```suggestion
  const QStringList parts =  QKeySequence( match.captured( 1 ) ).toString().split( "," );
   if ( std::all_of( elts.constBegin(), elts.constEnd(), []( const QString& part){
       return !part.trimmed().isEmpty();
   })

**Review by troopa81 — APPROVED** [review]

**Pull request merged**